src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,205 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+
+    static class Node<T> {
+        private T value;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.value = value;
+            this.prev = prev;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(tail, value, null);
+
+        if (head == null) {
+            head = newNode;
+        } else {
+            tail.next = newNode;
+        }
+        tail = newNode;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size()) {
+            throw new IndexOutOfBoundsException("Indeks jest poza zakresem listy: " + index);
+        }
+
+        if (index == 0) {
+            Node<T> newNode = new Node<>(null, value, head);
+            if (head != null) {
+                head.prev = newNode;
+            }
+            head = newNode;
+            if (tail == null) {
+                tail = newNode;
+            }
+        } else {
+            Node<T> current = head;
+            int currentIndex = 0;
+            while (current.next != null && currentIndex < index - 1) {
+                current = current.next;
+                currentIndex++;
+            }
+            if (current.next == null) {
+                Node<T> newNode = new Node<>(current, value, null);
+                current.next = newNode;
+                tail = newNode;
+            } else {
+                Node<T> newNode = new Node<>(current, value, current.next);
+                current.next.prev = newNode;
+                current.next = newNode;
+            }
+        }
```

Comment:
a bit too bulky, try to simplify (for example, using add(value) when size == index) or extract to private methods